File: rotate_image.c
#include<stdio.h>

/*
Algo:
- do a transpose of the matrix
- swap row 1 with row 3
*/

int transpose(int);
int swap_rows(int);

void main(){
	int arr[3][3] = {0,1,2,3,4,5,6,7,8}; // array initialization
	int n = sizeof(arr)/sizeof(arr[0][0]); // length of 2D array
	printf("%d\n", n); 

	int arrr = transpose(arr);
	//arr = swap_rows(arr);
}

int transpose(int a){
	int t_a[3][3] = {1,2,3,4,5,6,7,8,9};
	return t_a;
}

int swap_rows(int a){
	int s_a[3][3] = {1,2,3,4,5,6,7,8,9};
	return s_a;
}

File: rotate_image.py
#!/usr/bin/python

init_list = [[1,2,3],[4,5,6],[7,8,9]]

def rotate_matrix(matrix):
	'''Algo: 
	   	First: take transpose of the matrix
		Second: interchange row 0 and row 2
	'''

	matrix = zip(*matrix)
	return swap_rows(matrix)

def swap_rows(matrix):
	print matrix	
	row0, row2 = matrix[0], matrix[2]

	matrix[0] = row2
	matrix[2] = row0

	return matrix 

final_matrix = rotate_matrix(init_list)
print final_matrix

